### PR TITLE
Introduce Protobuf Message Handling for DRAIN Service.

### PR DESCRIPTION
**Pull request by AmartC**
This PR introduces logic to handle Protobuf messages within the Inferencing Service.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -4,8 +4,10 @@ WORKDIR /code
 COPY ./drain-service/drain_training_inferencing.py .
```
Comment (tybalex) [inline Dockerfile]: copy the dir

File: Dockerfile [hunk 2]
```diff
@@ -1,11 +1,8 @@
...
-WORKDIR /code
 
-COPY ./drain-service/drain_training_inferencing.py .
-COPY ./drain-service/drain_pretrained_inferencing.py .
-COPY ./drain-service/drain_modules.py .
-COPY ./drain-service/drain3.ini .
-COPY ./drain-service/drain3_control_plane_model_v0.4.1.bin .
-ADD ./drain-service/drain3 /code/drain3
+COPY ./drain-service/ /app/
+RUN pip install opni-aiops-apis==0.5.4
```
Comment (dbason) [inline Dockerfile]: This should be done in the base image

File: drain-service/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,6 @@
+boto3==1.17.45
+botocore==1.20.45
+cachetools==4.2.1
+jsonpickle==1.5.1
```
Comment (tybalex) [inline drain-service/requirements.txt:4]: again, I don't see jsonpickle being used?

**Review by sanjay920 — CHANGES_REQUESTED** [review]
There are several operations in this code that are inefficient.

There are no pandas related bulk operations executed so there is no need to convert to a data frame. You are looping over each pandas row - the same can be achieved by looping over `PayloadList`

In addition to this when you loop through the df rows, you are:
* converting each row to dict
* adding fields manually by strings to the dict
* adding row dict to list of row dicts

If you loop through `PayloadList`, you can set the payload's fields directly.

File: drain-service/requirements.txt (hunk 3 above)
Comment (AmartC) [reply drain-service/requirements.txt:4]: https://github.com/rancher/opni-drain-service/blob/protobuf-drain/drain-service/drain3/template_miner.py#L79
It is used for loading the DRAIN model binary file.

**Review by tybalex — DISMISSED** [review]

**Review by sanjay920 — DISMISSED** [review]

**Review by sanjay920 — APPROVED** [review]

**Review by tybalex — APPROVED** [review]

**Pull request merged**